pkg/generic/httpthrift_codec.go

```diff
@@ -46,15 +46,29 @@ type HTTPRequest = descriptor.HTTPRequest
 type HTTPResponse = descriptor.HTTPResponse
 
 type httpThriftCodec struct {
-	svcDsc           atomic.Value // *idl
-	provider         DescriptorProvider
-	codec            remote.PayloadCodec
-	binaryWithBase64 bool
+	svcDsc              atomic.Value // *idl
+	provider            DescriptorProvider
+	codec               remote.PayloadCodec
+	binaryWithBase64    bool
+	enableDynamicgoReq  bool
+	enableDynamicgoResp bool
+	convOpts            conv.Options
 }
 
-func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec) (*httpThriftCodec, error) {
+func newHTTPThriftCodec(p DescriptorProvider, codec remote.PayloadCodec, opts ...Option) (*httpThriftCodec, error) {
 	svc := <-p.Provide()
-	c := &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	var c *httpThriftCodec
+	if opts == nil {
+		c = &httpThriftCodec{codec: codec, provider: p, binaryWithBase64: false}
+	} else {
+		// codec with dynamicgo
+		gOpts := NewOptions(opts)
+		binaryWithBase64 := false
```

Comment:
`enableDynamicgo = !gOpts.dynamicgoConvOpts.NoBase64Binary` is enough

Reply:
gOpts -> dygOpts?

Reply:
@YangruiEmma Is `enableDynamicgo = !gOpts.dynamicgoConvOpts.NoBase64Binary` ok when users want to use dynamicgo conversion and set `gOpts.dynamicgoConvOpts.NoBase64Binary` true intentionally via `WithCustomDynamicgoConvOpts`?

Reply:
@YangruiEmma Can I add `isSetDynamicgoConvOpts` to generic.Options to check if we can enable dynamicgo instead?